engines/sci/sci.cpp

```diff
@@ -715,6 +715,63 @@ void SciEngine::runGame() {
 	} while (true);
 }
 
+// When the SCI engine enters an error state, this block will add additional VM engine context for error reporting
+void SciEngine::errorString(const char *buf_input, char *buf_output, int buf_output_size) {
+	EngineState *s = _gamestate;
+	Script *sci = s->_segMan->getScriptIfLoaded(s->xs->addr.pc.getSegment());
+
+	// If a script is actively loaded at the time of error.
+	if (sci) {
+		// Query the top-most stack frame even if it's not committed yet within the VM cycle.
+		const ExecStack *call = &(s->_executionStack.back());
+		Kernel *k = g_sci->getKernel();
+
+		// Note: if we are too early in the initialization process, this may not be populated yet.
+		const reg_t regVersion = s->variables[VAR_GLOBAL][kGlobalVarVersionNew];
+		const Common::String version = (regVersion.getOffset() > 0) ? ", Version: " + s->_segMan->getString(regVersion) : "";
+
+		const char *objname = s->_segMan->getObjectName(call->sendp);
+		Common::String callType;
+		Common::String callingFunc;
+		Common::String pcStr;
+
+		switch (call->type) {
+		case EXEC_STACK_TYPE_CALL: // Normal function
+			callType = "selector";
+			callingFunc += Common::String::format("%s::%s", objname, k->getSelectorName(call->debugSelector).c_str());
+			pcStr = Common::String::format("%04x:%04x", PRINT_REG(call->addr.pc));
+			break;
+		case EXEC_STACK_TYPE_KERNEL: // Kernel function
+			if (call->debugKernelSubFunction == -1){
+				callType = "kernel";
+				callingFunc += Common::String::format("k%s(", k->getKernelName(call->debugKernelFunction).c_str());
+			} else {
+				callType = "subkernel";
+				callingFunc += Common::String::format("k%s(", k->getKernelName(call->debugKernelFunction, call->debugKernelSubFunction).c_str());
+			}
+			pcStr = "none";
+			break;
+		default:
+			break;
+		}
+
+		Common::String errorStr = Common::String::format("Game: %s-%s%s\nRoom: %03d, Script: %03d, %s: %s(params if possible) @ pc=%s",
```

Comment:
The "params" bit should either be implemented, or removed as a placeholder from the text. It can be added in a comment as a TODO

Reply:
Ok, I’ll follow up with a TODO comment. Thanks for the clearer wording. I appreciate your help and expertise.